js/coinsbit.js

```diff
@@ -0,0 +1,799 @@
+        //            high: '47549.99',
+        //            low: '45571.08',
+        //            last: '46459.99',
+        //            volume: '8372.29022355',
+        //            deal: '388090406.73607745',
+        //          },
+        //          code: "200"
+        //     }
+        ticker = this.safeValue (response, 'result', {});
+        return this.parseTicker (ticker, market);
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        const request = params;
+        await this.loadMarkets ();
+        let tickers = {};
+        const response = await this.publicGetTickers (request);
+        //     {
+        //          success: true,
+        //          message: "",
+        //          result: {
+        //            BTC_USDT: {
+        //               at: '1640028811',
+        //               ticker: {
+        //                 bid: '46448.49682539',
+        //                 ask: '46458.41093569',
+        //                 open: '46687.39',
+        //                 high: '47550',
+        //                 low: '45584.27',
+        //                 last: '46462.77',
+        //                 vol: '32.56275846',
+        //                 deal: '1510574.62551143',
+        //                 change: '-0'
+        //               }
+        //            },
+        //            ...
+        //          },
+        //          code: "200"
+        //     }
+        tickers = this.safeValue (response, 'result', {});
+        const result = [];
+        const values = Object.values (tickers || []);
+        const keys = Object.keys (tickers);
+        for (let i = 0; i < values.length; i++) {
+            const key = keys[i];
+            const market = this.market (key);
+            result.push (this.extend (this.parseTicker (values[i], market), params));
+        }
+        return this.filterByArray (result, 'symbol', symbols);
+    }
+
+    parseTicker (ticker, market = undefined) {
+        //       at: '1640028811',
+        //       ticker: {
+        //         bid: '46448.49682539',
+        //         ask: '46458.41093569',
+        //         open: '46687.39',
+        //         high: '47550',
+        //         low: '45584.27',
+        //         last: '46462.77',
+        //         vol: '32.56275846',
+        //         deal: '1510574.62551143',
+        //         change: '-0'
+        //       }
+        let tickerFinal = {};
+        let timestamp = null;
+        if ('ticker' in ticker) { // if arrived from fetchTickers
+            tickerFinal = this.safeValue (ticker, 'ticker', {});
+            timestamp = this.safeTimestamp (ticker, 'at');
+        } else { // if arrived from fetchTicker, then there is nothing to be done more
+            tickerFinal = ticker;
+            timestamp = this.milliseconds ();
+        }
+        const last = this.safeNumber (tickerFinal, 'last');
+        return this.safeTicker ({
+            'symbol': market['symbol'],
```

Comment:
Before we can dereference `market['symbol']` we must make sure `market` is not `undefined`.

Therefore we usually have a line like this:

```JavaScript
parseTicker (ticker, market) {
    // ...
    // const marketId = this.safeString (ticker, 'market'); // for example
    // market = this.safeMarket (marketId, market, delimiter); // in general case
    market = this.safeMarket (undefined, market); // in this case
    // ...
    return this.safeTicker ({
        'symbol': market['symbol'], // it's safe after safeMarket
        // ...
    });
}
```